kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieContainerInstanceImpl.java

```diff
@@ -84,16 +114,49 @@ public KieContainerResource getResource() {
         return resource;
     }
 
+    public InternalKieScanner getScanner() {
+        return this.scanner;
+    }
+
     public void setResource(KieContainerResource resource) {
         this.resource = resource;
     }
 
     public void setScanner(InternalKieScanner scanner) {
         this.scanner = scanner;
+        // we also need to update the underlaying scanner resource to avoid inconsistency
+        if (scanner != null) {
+            KieScannerStatus newStatus = KieContainerInstanceImpl.mapScannerStatus(scanner.getStatus());
+            long newPollingInterval = scanner.getPollingInterval();
+            resource.setScanner(new KieScannerResource(newStatus, newPollingInterval));
+        } else {
+            resource.setScanner(new KieScannerResource(KieScannerStatus.DISPOSED));
+        }
     }
 
-    public InternalKieScanner getScanner() {
-        return this.scanner;
+    public void startScanner(long pollingInterval) {
+        if (this.scanner == null) {
+            throw new IllegalStateException("Can not start non-existing (null) scanner!");
+        }
+        this.scanner.start(pollingInterval);
+        this.getResource().setScanner(new KieScannerResource(KieScannerStatus.STARTED, pollingInterval));
+    }
+
+    public void stopScanner() {
+        if (this.scanner == null) {
+            throw new IllegalStateException("Can not stop non-existing (null) scanner!");
+        }
+        this.scanner.stop();
+        this.getResource().getScanner().setStatus(KieScannerStatus.STOPPED);
+    }
+
+    public void disposeScanner() {
+        if (this.scanner == null) {
+            throw new IllegalStateException("Can not dispose non-existing (null) scanner!");
+        }
+        this.scanner.shutdown();
+        this.scanner = null;
+        this.getResource().setScanner(new KieScannerResource(KieScannerStatus.DISPOSED));
```

Comment:
Yes, that should be possible. I added just those needed to fix the sync issues, but this would make it more consistent, so I'll add those as well. I'll also see if we could remove the getScanner() method to avoid any possible mutation by 3rd party objects.